lib/tacos/plan

```diff
@@ -7,11 +7,47 @@ exec 3>&1 1>&2
 # note: for terragrunt, tfplan must be absolute
 TACOS_TFPLAN="${TACOS_TFPLAN:-$PWD/tfplan}"
 TACOS_LOCK="${TACOS_LOCK:-false}"
+cache=.config/tf-lock-info
+
+#todo: make this an artifact instead
+# we expect at most 6 slices to be fighting
+git_push_aggressively() {
+  sleep=3
+  limit=20 # 3 min 
+  while ! (set -ex; git push -q); do
+    if (( sleep > limit )); then exit 1; fi
+    echo "failed! trying again after $sleep seconds..."
+    sleep $sleep
+    (( sleep += 3 ))
+    (set -ex
+    git pull --rebase origin "$GITHUB_HEAD_REF" 
+    )
+  done
+}
 
 if "$TACOS_LOCK"; then
   ( set -ex
     env GETSENTRY_SAC_VERB=state-admin sudo-gcp tf-lock-acquire
   )
+  if [[ -d $cache ]]; then
+    git add -f $cache
+    if ! git diff --cached --exit-code $cache; then
+      git config --global user.email "$USER@$HOSTNAME"
+      git config --global user.name "$USER"
+      git config --global push.default current
+      #git checkout -b "$GITHUB_HEAD_REF"
+      ( set -ex
+      git fetch origin --depth=1 "$GITHUB_HEAD_REF"
+      git checkout "$GITHUB_HEAD_REF"
+      # graft="$(git rev-parse HEAD)"
+        # add .terraform-lock changes 
+        git add -u .
+        git status
+        git commit -m "updating tf-lock cache: $TF_ROOT_MODULE"
+      )
+      git_push_aggressively
+    fi
+  fi
```

Comment:
this is all incidental complexity with respect to "tacos plan". Let's move it to its own file. Perhaps `lib/tacos/lib/handle-tflock-cache`. That path is on the PATH when this file is running (pretty sure) so you can run it with simply `handle-tflock-cache`.